File: server/backend/app/models/category.py
from sqlalchemy import Column, Integer, String, ForeignKey, DateTime
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from ..database import Base

class Category(Base):
    __tablename__ = "category"

    category_id = Column(Integer, primary_key=True, index=True)
    name = Column(String(255), nullable=False, index=True)
    warehouse_id = Column(Integer, ForeignKey("warehouse.warehouse_id"), nullable=False)
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())

    warehouse = relationship("Warehouse", back_populates="categories")
    items = relationship("Item", back_populates="category")

File: app/lib/models/category.dart
class Category {
  final int categoryId;
  final String name;
  final int warehouseId;

  Category({
    required this.categoryId,
    required this.name,
    required this.warehouseId,
  });

  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      categoryId: json['category_id'],
      name: json['name'],
      warehouseId: json['warehouse_id'],
    );
  }
}
